app/presenters/order_detail_notice_presenter.rb

```diff
@@ -0,0 +1,85 @@
+class OrderDetailNoticePresenter < DelegateClass(OrderDetail)
+
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  def statuses
+    [].tap do |statuses|
+      statuses << Notice.new(:in_review) if in_review?
+      statuses << Notice.new(:in_dispute) if in_dispute?
+      statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
+      statuses << Notice.new(:in_open_journal) if in_open_journal?
+    end
+  end
+
```

Comment:
I mildly prefer the local variable version to the tap version, but if you don't have a problem reading the tap version, don't change it on my account.